shell/platform/fuchsia/flutter/platform_view.cc

```diff
@@ -35,6 +35,76 @@ static constexpr char kAccessibilityChannel[] = "flutter/accessibility";
+static constexpr char kTextInputActionSearch[] = "TextInputAction.search";
+static constexpr char kTextInputActionSend[] = "TextInputAction.send";
+static constexpr char kTextInputActionUnspecified[] =
+    "TextInputAction.unspecified";
+
+// Converts Flutter TextInputAction to Fuchsia action enum.
+static fuchsia::ui::input::InputMethodAction IntoInputMethodAction(
+    const std::string action_string) {
+  if (action_string == kTextInputActionNewline) {
+    return fuchsia::ui::input::InputMethodAction::NEWLINE;
+  } else if (action_string == kTextInputActionDone) {
+    return fuchsia::ui::input::InputMethodAction::DONE;
+  } else if (action_string == kTextInputActionGo) {
+    return fuchsia::ui::input::InputMethodAction::GO;
+  } else if (action_string == kTextInputActionNext) {
+    return fuchsia::ui::input::InputMethodAction::NEXT;
+  } else if (action_string == kTextInputActionPrevious) {
+    return fuchsia::ui::input::InputMethodAction::PREVIOUS;
+  } else if (action_string == kTextInputActionNone) {
+    return fuchsia::ui::input::InputMethodAction::NONE;
+  } else if (action_string == kTextInputActionSearch) {
+    return fuchsia::ui::input::InputMethodAction::SEARCH;
+  } else if (action_string == kTextInputActionSend) {
+    return fuchsia::ui::input::InputMethodAction::SEND;
+  } else if (action_string == kTextInputActionUnspecified) {
+    return fuchsia::ui::input::InputMethodAction::UNSPECIFIED;
+  }
+  // Substituting DONE for an unexpected action string will probably be OK.
+  FML_LOG(WARNING) << "unexpected action_string: " << action_string;
+  return fuchsia::ui::input::InputMethodAction::DONE;
+}
+
+// Converts the Fuchsia action enum into Flutter TextInputAction.
+static const std::string IntoTextInputAction(
+    fuchsia::ui::input::InputMethodAction action) {
+  if (action == fuchsia::ui::input::InputMethodAction::NEWLINE) {
+    return kTextInputActionNewline;
+  } else if (action == fuchsia::ui::input::InputMethodAction::DONE) {
+    return kTextInputActionDone;
+  } else if (action == fuchsia::ui::input::InputMethodAction::GO) {
+    return kTextInputActionGo;
+  } else if (action == fuchsia::ui::input::InputMethodAction::NEXT) {
+    return kTextInputActionNext;
+  } else if (action == fuchsia::ui::input::InputMethodAction::PREVIOUS) {
+    return kTextInputActionPrevious;
+  } else if (action == fuchsia::ui::input::InputMethodAction::NONE) {
+    return kTextInputActionNone;
+  } else if (action == fuchsia::ui::input::InputMethodAction::SEARCH) {
+    return kTextInputActionSearch;
+  } else if (action == fuchsia::ui::input::InputMethodAction::SEND) {
+    return kTextInputActionSend;
+  } else if (action == fuchsia::ui::input::InputMethodAction::UNSPECIFIED) {
+    return kTextInputActionUnspecified;
+  }
+  // Substituting "done" for an unexpected text input actoin will probably
+  // be OK.
+  FML_LOG(WARNING) << "unexpected action: " << static_cast<uint32_t>(action);
```

Comment:
also worried here about log spam risk

Reply:
See prior comment, I think it's the same discussion.